jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditor.kt

```diff
@@ -34,25 +36,26 @@ class QueryEditor internal constructor(
     lateinit var retrieveSavedQueriesButton: JButton
     private lateinit var tablePanel: SimpleToolWindowPanel
     lateinit var queryBox: JTextArea
-    lateinit var logGroupLabel: JLabel
     lateinit var endDate: DatePicker
     lateinit var queryEditorBasePanel: JPanel
     lateinit var relativeTimeUnit: JComboBox<Pair<String, ChronoUnit>>
+    private lateinit var numberFormat: NumberFormat
     lateinit var relativeTimeNumber: JFormattedTextField
     lateinit var startDate: DatePicker
-    private lateinit var showLogGroupTable: AddRemoveLogGroupTable
-
-    private fun initArLogGroupTable() {
-        showLogGroupTable.tableView.listTableModel
-        showLogGroupTable.getSelLogGroups()
-    }
+    lateinit var queryGroupScrollPane: JBScrollPane
+    lateinit var logGroupTable: LogGroupSelectorTable
+    private lateinit var timePanel: JPanel
+    private lateinit var searchPanel: JPanel
 
     private fun createUIComponents() {
         tablePanel = SimpleToolWindowPanel(false, true)
-        showLogGroupTable = AddRemoveLogGroupTable(project)
-        initArLogGroupTable()
-        // tablePanel.setContent(showLogGroupTable.component)
-        relativeTimeNumber = JFormattedTextField(NumberFormat.getIntegerInstance())
+        logGroupTable = LogGroupSelectorTable()
+        tablePanel.setContent(logGroupTable.component)
+        // lateinit since this method runs before the standard initialization flow
+        numberFormat = NumberFormat.getIntegerInstance()
+        relativeTimeNumber = JFormattedTextField(numberFormat)
+        // arbitrary length
```

Comment:
arbitrary length? what does this do?

Reply:
is this the thing that drives the "minutes/hours etc" box - if so 15 columns seems MASSIVE - could prolly easily get away with 5 or even 3.

Reply:
comments didn't publish, but yes